docs/testing-with-codeception.md

```diff
@@ -0,0 +1,746 @@
+
+```php
+/**
+  * @example ["/api/", 200]
+  * @example ["/api/protected", 401]
+  * @example ["/api/not-found-url", 404]
+  * @example ["/api/faulty", 500]
+  */
+public function test_api_responses( ApiTester $I, \Codeception\Example $example ) {
+    $I->sendGet( $example[0] );
+    $I->seeResponseCodeIs( $example[1] );
+}
+```
+
+You can define examples in Doctrine or JSON style, eg: `@example ["/api/", 200]` or `@example { "url": "/api/", "code": 200 }` or `@example(url="/api/", code=200)`.
+
+#### DataProviders
+
+You can also use  PHPUnit's `@dataProvider` pattern to create _dynamic data sets_ for test methods, where the test will run once per each data set returned from the protected data provider method, but the syntax differs a bit given the way test methods are written, eg:
+
+```php
+/**
+* @dataProvider pageProvider
+*/
+public function testStaticPages( AcceptanceTester $I, \Codeception\Example $example ) {
+    $I->amOnPage( $example['url'] );
+    $I->see( $example['title'], 'h1' );
+    $I->seeInTitle( $example['title'] );
+}
+
+/**
+ * @return array
+ */
+protected function pageProvider() {
+    return [
+        [ 'url' => "/", 'title' => "Welcome" ],
+        [ 'url' => "/info", 'title' => "Info" ],
+        [ 'url' => "/about", 'title' => "About Us" ],
+        [ 'url' => "/contact", 'title' => "Contact Us" ]
+    ];
+}
+```
+
+[Read more on Codeception docs](https://codeception.com/docs/07-AdvancedUsage.html#Example-Annotation).
+
+#### Before and After
+
+Codeception tests have special annotation types to execute methods before a certain test method, where you can define one or more prerequisites/cleanup functions, eg:
+
+```php
+protected function activate( AcceptanceTester $I ) {
+    $this->loginAsAdmin();
+    $this->activatePlugin( 'some-plugin' );
+}
+
+protected function cleanup( AcceptanceTester $I ) {
+    $I->deactivatePlugin( 'some-plugin' );
+    $I->logout();
+}
+
+/**
+ * @before activate
+ * @before anotherPrerequisite
+ * @after cleanup
+ */
+public function checkPluginPageExists( AcceptanceTester $I ) {
+    // ...
+}
+```
+
+#### Environment
+
+Codeception tests can be instructed to run in multiple / different environments, via the `@env` special annotation, eg:
+
+```php
+/**
+ * @env chrome
+ * @env firefox
+ */
+public function someTest() {}
+```
+
+### Available modules
+
+Altis' Codeception integration comes bundled with `wp-browser` library, which provides additional modules to simplify testing WordPress applications. Altis pre-configures such modules via the zero-config installation, so you don't need to manually configure them unless you need to override some of the default values, which you can do via test suite configuration, eg:
```

Comment:
```suggestion
Altis' Codeception integration comes bundled with the [`wp-browser` library](https://github.com/lucatume/wp-browser), which provides additional modules to simplify testing WordPress applications. Altis pre-configures these modules via the zero-config installation, so you don't need to manually configure them unless you need to override some of the default values, which you can do via test suite configuration, eg:
```
Keep an eye on these long sentences and consider breaking them up a bit to make things clearer.